src/org/sosy_lab/java_smt/solvers/z3/Z3FloatingPointFormulaManager.java

```diff
@@ -70,6 +71,30 @@ protected Long makeNumberImpl(double pN, FloatingPointType pType, Long pRounding
     return makeNumberImpl(Double.toString(pN), pType, pRoundingMode);
   }
 
+  @Override
+  protected Long makeNumberImpl(
+      BigInteger exponent, BigInteger mantissa, boolean signBit, FloatingPointType type) {
+
+    long signBv = Native.mkBvNumeral(z3context, 1, new boolean[] {signBit});
+    long expoBv =
+        Native.mkNumeral(
+            z3context, exponent.toString(), Native.mkBvSort(z3context, type.getExponentSize()));
+    long mantBv =
+        Native.mkNumeral(
+            z3context, mantissa.toString(), Native.mkBvSort(z3context, type.getMantissaSize()));
+
+    assert Native.getBvSortSize(z3context, Native.getSort(z3context, signBv)) == 1
```

Comment:
Note that currently, that test fails for Z3, because it overwrites the created constant's sort. I don't know how to solve this problem, I suspect a bug in Z3. If I remove line 82, then signBv's size remains 1, but as-is, its size is 23 (= mantissa size).

Reply:
It does not matter which constructor method I use on z3's side (`string` or `bool[]`).

Reply:
Nevermind, I was just missing some incRefs and decRefs.